src/compass/s1_geocode_slc.py

```diff
@@ -79,16 +82,34 @@ def run(cfg: GeoRunConfig):
             if cfg.rdr2geo_params.geocode_metadata_layers:
                s1_geocode_metadata.run(cfg, fetch_from_scratch=True)
 
+
+        #Load the input burst SLC
+        temp_slc_path = f'{scratch_path}/{id_pol}_temp.vrt'
+        burst.slc_to_vrt_file(temp_slc_path)
+
+        # Check if EAP correction is necessary
+        check_eap = is_eap_correction_necessary(burst.ipf_version)
+        if check_eap.phase_correction:
+            temp_slc_path_corrected = temp_slc_path.replace('_temp.vrt',
+                                                            '_corrected_temp.rdr')
+            apply_eap_correction(burst,
```

Comment:
@LiangJYu Good point. Maybe we should pass `isce3.io.Raster` objects, rather than path to the rasters as `str`. Same for `range_split_spectrum` that follows the EAP correction.
For that I would need to figure out how to have read/write access to the data in ISCE3 raster. Please create an issue for that, and let's discuss about the operations there.